app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,185 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+@DisplayName("TaskService 클래스")
+class TaskServiceTest {
+    final String givenTitle = "sample";
+    final long givenID = 1L;
+
+    TaskService emptySubject() {
+        return new TaskService();
+    }
+
+    TaskService existsSubject() {
+        Task source = new Task();
+        source.setTitle(givenTitle);
+
+        TaskService taskService = new TaskService();
+        taskService.createTask(source);
+
+        return taskService;
+    }
+
+    @Nested
+    @DisplayName("getTasks 메서드는")
+    class Describe_getTasks {
+
+        @Nested
+        @DisplayName("tasks 가 없을 때")
+        class Context_without_tasks {
+
+            @Test
+            @DisplayName("빈 ArrayList 를 리턴한다.")
```

Comment:
`빈` 이라고 하면 어떤 사람은 `Bean`으로 착각할 수도 있습니다. 하하 설마 그럴 리는 없겠지만... 가급적이면 2가지 의미로 해석되지 않도록 작성하는 것도 생각보다 굉장한 실력 포인트입니다.

한편, 리턴 타입이 `ArrayList`인지를 검사하는 것 같은 테스트 메시지는 아쉽습니다. 왜냐하면 준희님이 작성하신 메소드를 읽어보면 리턴 타입이 `ArrayList`가 아닙니다.
![image](https://user-images.githubusercontent.com/1855714/106604891-98a5a800-65a3-11eb-9d77-096e6e61c465.png)

다음과 같이 수정하는 것을 생각해 볼 수 있을 것입니다.
```suggestion
            @DisplayName("비어 있는 리스트를 리턴한다.")
```

한편, 수학적인 측면을 고려해 좀 더 추상화하고 싶다면 다음과 같이 하는 것도 생각할 수 있습니다. list 가 ordered set 이라는 것을 의식한 메시지입니다. 하지만 개발하고 있는 프로젝트에서 집합과 리스트의 추상적인 관계가 중요하지 않게 다뤄지고 있다면 굳이 이렇게 할 필요는 없습니다.
```suggestion
            @DisplayName("비어 있는 집합을 리턴한다.")
```